src/ProductConstructionService/ProductConstructionService.Api/Queue/JobsProcessor.cs

```diff
@@ -28,15 +28,15 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
         _logger.LogInformation("Starting to process PCS jobs {queueName}", _options.Value.JobQueueName);
         while (!cancellationToken.IsCancellationRequested)
         {
-            using (_scopeManager.BeginJobScopeWhenReady())
+            using (JobScope jobScope = _scopeManager.BeginJobScopeWhenReady())
             {
                 if (cancellationToken.IsCancellationRequested)
                 {
                     break;
                 }
                 try
                 {
-                    await ReadAndProcessJobAsync(queueClient, cancellationToken);
+                    await ReadAndProcessJobAsync(queueClient, jobScope, cancellationToken);
                 }
                 // If the cancellation token gets cancelled, we just want to exit
                 catch (OperationCanceledException)
```

Comment:
The way it was changed, this one might not leak out as you catch everything inside `ReadAndProcessJobAsync`.
Maybe you should rethrow `OperationCanceledException` from there?

Reply:
I added rethrowing `OperationCanceledException` from `ReadAndProcessJobAsync`. we'll need to be careful how we handle the throwing inside of the jobs once we start implementing them

Reply:
Oh yeah, you're right. Maybe this wasn't such a good idea after all. Now I am thinking if we had unrelated cancellation bubbling out, it will make the job listener stop processing